test/its.js

```diff
@@ -0,0 +1,365 @@
+                arguments: [objectIds.tokenId],
+            });
+
+            await txb.moveCall({
+                target: `${deployments.example.packageId}::its_example::deploy_remote_interchain_token`,
+                arguments: [
+                    objectIds.its,
+                    objectIds.gateway,
+                    objectIds.gasService,
+                    trustedSourceChain,
+                    TokenId,
+                    gas,
+                    '0x',
+                    deployer.toSuiAddress(),
+                ],
+            });
+
+            await txb.signAndExecute(deployer);
+        });
+
+        it('should receive interchain token deployment from other chain successfully', async () => {
+            // Deploy the interchain token and store object ids for treasury cap and metadata
+            const publishReceipt = await publishPackage(client, deployer, 'interchain_token');
+            const treasuryCap = findObjectId(publishReceipt.publishTxn, 'TreasuryCap');
+            const metadata = findObjectId(publishReceipt.publishTxn, 'CoinMetadata');
+
+            // Approve ITS transfer message
+            const messageType = 1; // MESSAGE_TYPE_DEPLOY_INTERCHAIN_TOKEN
+            const tokenId = hexlify(randomBytes(32)); // The token ID to transfer
+            const name = toUtf8Bytes('Quote');
+            const symbol = toUtf8Bytes('Q');
+            const decimals = 9;
+            const distributor = '0x';
+
+            // ITS transfer payload from Ethereum to Sui
+            const payload = defaultAbiCoder.encode(
+                ['uint256', 'uint256', 'bytes', 'bytes', 'uint256', 'bytes'],
+                [messageType, tokenId, name, symbol, decimals, distributor],
+            );
+
+            const message = {
+                source_chain: trustedSourceChain,
+                message_id: hexlify(randomBytes(32)),
+                source_address: trustedSourceAddress,
+                destination_id: objectIds.itsChannel,
+                payload_hash: keccak256(payload),
+            };
+
+            await approveMessage(client, keypair, gatewayInfo, message);
+
+            const txb = new TxBuilder(client);
+
+            const approvedMessage = await txb.moveCall({
+                target: `${deployments.axelar_gateway.packageId}::gateway::take_approved_message`,
+                arguments: [
+                    objectIds.gateway,
+                    message.source_chain,
+                    message.message_id,
+                    message.source_address,
+                    message.destination_id,
+                    payload,
+                ],
+            });
+
+            txb.moveCall({
+                target: `${deployments.its.packageId}::service::give_unregistered_coin`,
+                arguments: [objectIds.its, treasuryCap, metadata],
+                typeArguments: [`${publishReceipt.packageId}::q::Q`],
+            });
+
+            await txb.moveCall({
```

Comment:
As part of ITS operations task, add a script to use the ITS example via the deployments repo as well

Reply:
will do this later